colossalai/fx/codegen/activation_checkpoint_codegen.py

```diff
@@ -566,9 +576,13 @@ def emit_node(node: Node):
             orig_args.insert(0, 'self')
         code = ''.join(body)
         code = '\n'.join('    ' + line for line in code.split('\n'))
+
+        # as we need colossalai.utils.checkpoint, we need to import colossalai
+        # in forward function
         fn_code = f"""
 {wrap_stmts}
 
 def forward({', '.join(orig_args)}){maybe_return_annotation[0]}:
+    import colossalai
```

Comment:
`import` should not occur in the forward function. One way is to make sure `colossalai` is imported during tracing activation checkpointing.

Reply:
This should be fixed in a separate PR.